src/Microsoft.AspNet.Mvc.Razor/Compilation/RoslynCompilationService.cs

```diff
@@ -300,5 +322,156 @@ private static string ReadFileContentsSafely(IFileProvider fileProvider, string
 
             return null;
         }
+
+        public class WalkerTexasRanger : CSharpSyntaxWalker
+        {
+            public WalkerTexasRanger(SemanticModel semanticModel)
+            {
+                SemanticModel = semanticModel;
+            }
+
+            private SemanticModel SemanticModel { get; }
+
+            public List<SimpleLambdaExpressionSyntax> Expressions { get; } = new List<SimpleLambdaExpressionSyntax>();
+
+            public override void VisitSimpleLambdaExpression(SimpleLambdaExpressionSyntax node)
+            {
+                var type = SemanticModel.GetTypeInfo(node);
+                if (type.ConvertedType.Name != "Expression" && type.ConvertedType.ContainingAssembly.Name != "System.Core")
+                {
+                    return;
+                }
+
+                if (!node.Parent.IsKind(SyntaxKind.Argument))
+                {
+                    return;
+                }
+
+                var parameter = node.Parameter;
+                if (IsValidForHoisting(parameter, node.Body))
+                {
+                    Expressions.Add(node);
+                }
+            }
+
+            private bool IsValidForHoisting(ParameterSyntax parameter, CSharpSyntaxNode node)
+            {
+                if (node.IsKind(SyntaxKind.IdentifierName))
+                {
+                    var identifier = (IdentifierNameSyntax)node;
+                    if (identifier.Identifier.Text == parameter.Identifier.Text)
+                    {
+                        return true;
+                    }
+                }
+                else if (node.IsKind(SyntaxKind.SimpleMemberAccessExpression))
+                {
+                    var memberAccess = (MemberAccessExpressionSyntax)node;
+                    var lhs = memberAccess.Expression;
+                    return IsValidForHoisting(parameter, lhs);
+                }
+
+                return false;
+            }
+        }
+
+        public class JamesTrivette : CSharpSyntaxRewriter
```

Comment:
Yeah, I dunno. It would feel wrong giving top billing to either CD or Alex either. Also, Jimmy **did** play for the Cowboys before that career ending injury.

I realize now you don't really need a separate pass for the walker, you could do all of this in one pass.. which would have avoided all of this.

Also, @Eilon - would your uncle appear as a special guest at one of our team meetings?